VideoPlayer/tests/Video-specs.js

```diff
@@ -27,237 +20,250 @@ describe('VideoPlayer.Video', () => {
+			);
 
-			const actual = getSourceNode(subject);
+			const actual = screen.getByTestId('video-id');
 
 			expect(actual).toBe(expected);
 		});
 
 		test('should use the same node when changing `preloadSource`', () => {
-			const subject = mount(
-				<Video source="abc.mp4" preloadSource="def.mp4" />
+			const {rerender} = render(
+				<Video data-testid="video-id" preloadSource="def.mp4" source="abc.mp4" />
 			);
 
-			const expected = getPreloadNode(subject);
+			const expected = screen.getByTestId('video-id').nextElementSibling;
 
-			subject.setProps({
-				preloadSource: 'ghi.mp4'
-			});
+			rerender(
+				<Video data-testid="video-id" preloadSource="ghi.mp4" source="abc.mp4" />
+			);
 
-			const actual = getPreloadNode(subject);
+			const actual = screen.getByTestId('video-id').nextElementSibling;
 
 			expect(actual).toBe(expected);
 		});
 
 		test('should swaps nodes when swapping `source` and `preloadSource`', () => {
-			const subject = mount(
-				<Video source="abc.mp4" preloadSource="def.mp4" />
+			const {rerender} = render(
+				<Video data-testid="video-id" preloadSource="def.mp4" source="abc.mp4" />
 			);
 
-			const source = getSourceNode(subject);
-			const preload = getPreloadNode(subject);
+			const source = screen.getByTestId('video-id');
+			const preload = screen.getByTestId('video-id').nextElementSibling;
 
-			subject.setProps({
-				source: 'def.mp4',
-				preloadSource: 'abc.mp4'
-			});
+			rerender(
+				<Video data-testid="video-id" preloadSource="abc.mp4" source="def.mp4" />
+			);
 
-			expect(getSourceNode(subject)).toBe(preload);
-			expect(getPreloadNode(subject)).toBe(source);
-		});
+			const newSource = screen.getByTestId('video-id');
+			const newPreload = screen.getByTestId('video-id').nextElementSibling;
 
+			expect(newSource).toBe(preload);
+			expect(newPreload).toBe(source);
+		});
 
 		test('should not swap nodes on re-render after swapping `source` and `preloadSource`', () => {
-			const subject = mount(
-				<Video source="abc.mp4" preloadSource="def.mp4" />
+			const {rerender} = render(
+				<Video data-testid="video-id" preloadSource="def.mp4" source="abc.mp4" />
 			);
 
-			const source = getSourceNode(subject);
-			const preload = getPreloadNode(subject);
+			const source = screen.getByTestId('video-id');
+			const preload = screen.getByTestId('video-id').nextElementSibling;
 
-			subject.setProps({
-				source: 'def.mp4',
-				preloadSource: 'abc.mp4'
-			});
+			rerender(
+				<Video data-testid="video-id" preloadSource="abc.mp4" source="def.mp4" />
+			);
+
+			const newSource = screen.getByTestId('video-id');
+			const newPreload = screen.getByTestId('video-id').nextElementSibling;
 
-			subject.setProps({});
+			rerender(
+				<Video />
+			);
 
-			expect(getSourceNode(subject)).toBe(preload);
-			expect(getPreloadNode(subject)).toBe(source);
+			expect(newSource).toBe(preload);
+			expect(newPreload).toBe(source);
```

Comment:
Here you are checking values saved before the new rerender. You will need to read the nodes again and compare them tp newSource and newPreload

Reply:
Thank you!